### PR TITLE
feat: collection to manage GoDaddy API

### PR DESCRIPTION
**Pull request by jacobdotcosta**

### CODE REVIEW

File: ansible/ansible_collections/snowdrop/godaddy/roles/dns_info/molecule/default/INSTALL.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+***********************************
+Delegated driver installation guide
```
Comment (cmoulliard) [inline ansible/ansible_collections/snowdrop/godaddy/roles/dns_info/molecule/default/INSTALL.rst:2]: same remark as before

File: ansible/ansible_collections/snowdrop/godaddy/roles/dns/molecule/default/INSTALL.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
+***********************************
+Delegated driver installation guide
```
[the remark this answers is not in the document]
Comment (jacobdotcosta) [reply ansible/ansible_collections/snowdrop/godaddy/roles/dns/molecule/default/INSTALL.rst:2]: It's a documentation file.

```quote
This file contains instructions for additional dependencies needed for a successful interaction between Molecule and the driver.
```

File: ansible/ansible_collections/snowdrop/godaddy/roles/dns_info/molecule/default/INSTALL.rst (hunk 1 above)
Comment (jacobdotcosta) [reply ansible/ansible_collections/snowdrop/godaddy/roles/dns_info/molecule/default/INSTALL.rst:2]: Improved the file contents.

File: ansible/ansible_collections/snowdrop/godaddy/roles/dns/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+| Parameter | DNS Description
+| --- | ---
+| `api_environment`<br/><span style="color:fuchsia">string</span> | GoDaddy API environment to use:<ul><li>_Empty/Default_: OTE</li><li>`production` or `prod`: Production</li></ul>
+| `state`<br/><span style="color:fuchsia">string</span><br/><span style="color:red">required</span> | State of the record: <ul><li>`present`</li><li>`absent`</li></ul>
+| `domain_name`<br/><span style="color:fuchsia">string</span><br/><span style="color:red">required</span> | GoDaddy domain to associate the DNS record with. 
+| `record_type`<br/><span style="color:fuchsia">string</span><br/><span style="color:red">required</span> | DNS type of the record
+| `record_name`<br/><span style="color:fuchsia">string</span><br/><span style="color:red">required</span> | Name of the DNS record
+| `dns`<br/><span style="color:fuchsia">json</span><br/><span style="color:red">required</span> | DNS data
+
+The contents of the `dns` variable are the following.
```
[the remark this answers is not in the document]
Comment (jacobdotcosta) [reply ansible/ansible_collections/snowdrop/godaddy/roles/dns/README.md]: fixed

**Review by cmoulliard — APPROVED** [review]
As this PR is really big (73 files) I will trust you

**Pull request merged**